controllers/storagecluster/exporter.go

```diff
@@ -209,9 +265,515 @@ func CreateOrUpdateServiceMonitor(r *StorageClusterReconciler, instance *ocsv1.S
 		return nil, err
 	}
 
-	err = r.Client.Update(context.TODO(), oldSm)
+	err = r.Client.Update(ctx, oldSm)
 	if err != nil {
 		return nil, fmt.Errorf("failed to update metrics exporter servicemonitor %v. %v", namespacedName, err)
 	}
 	return serviceMonitor, nil
 }
+
+func getMetricExporterDeployment(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) *appsv1.Deployment {
+	var (
+		falsePtr                      = new(bool) // defaults to 'false'
+		truePtr                       = new(bool)
+		progressDeadlineSeconds int32 = 600
+		replicas                int32 = 1
+		revisionHistoryLimit    int32 = 1
+		maxSurge                      = intstr.Parse("25%")
+		maxUnavailable                = intstr.Parse("25%")
+	)
+	*truePtr = true
+	dep := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      metricsExporterName,
+			Namespace: instance.Namespace,
+			OwnerReferences: []metav1.OwnerReference{{
+				APIVersion:         instance.APIVersion,
+				BlockOwnerDeletion: falsePtr,
+				Controller:         falsePtr,
+				Kind:               instance.Kind,
+				Name:               instance.Name,
+				UID:                instance.UID,
+			}},
+			Labels: map[string]string{
+				"app.kubernetes.io/component": metricsExporterName,
+				"app.kubernetes.io/name":      metricsExporterName,
+				"app.kubernetes.io/version":   "0.0.1",
+			},
+		},
+		Spec: appsv1.DeploymentSpec{
+			ProgressDeadlineSeconds: &progressDeadlineSeconds,
+			Replicas:                &replicas,
+			RevisionHistoryLimit:    &revisionHistoryLimit,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app.kubernetes.io/component": metricsExporterName,
+					"app.kubernetes.io/name":      metricsExporterName,
```

Comment:
Use `exporterLabels` instead of redefining these labels.

Reply:
Done